src/main/java/com/divelink/server/service/PracticeNoticeService.java

```diff
@@ -0,0 +1,181 @@
+      throw new IllegalArgumentException("장비 중복 선택 불가 (중복 gearId)");
+    }
+    List<Long> ids = new ArrayList<>(unique);
+
+    // 2) 집계 한 번으로 capacity/used/price 가져오기
+    List<GearCapacityRow> rows = gearRepository.checkCapacityForNotice(practiceNotice.getId(), ids);
+
+    if (rows.size() != ids.size()) {
+      throw new IllegalArgumentException("존재하지 않는 장비 ID 포함");
+    }
+
+    // 3) 재고 검증 + 총액 계산
+    int totalFee = 0;
+    for (GearCapacityRow r : rows) {
+      int remain = (int) (r.getCapacity() - r.getUsed());
+      if (remain <= 0) {
+        throw new IllegalArgumentException("장비 재고 부족: gearId=" + r.getGearId());
+      }
+      totalFee += r.getPrice(); // 각 장비 1개씩 가정
+    }
+
+    PracticeApplication application = PracticeApplication.builder()
+        .userId(userId)
+        .practiceNotice(practiceNotice)
+        .status(Status.PAYMENT_PENDING)
+        .totalFee(totalFee)
+        .appliedAt(LocalDateTime.now())
+        .gearIds(ids)
+        .build();
+    practiceApplicationRepository.save(application);
+  }
+
+  public PracticeApplicationResponse getApplicationInfo(Long practiceNoticeId, String userId) {
+    PracticeNotice practiceNotice = practiceNoticeRepository.findById(practiceNoticeId)
+        .orElseThrow(()-> new EntityNotFoundException("공지글 없음"));
+
+    PracticeApplication applicationInfo = practiceApplicationRepository.findByUserIdAndPracticeNotice(userId, practiceNotice)
+        .orElseThrow(()->new EntityNotFoundException("신청 내역을 찾을 수 없음"));
+
+    List<Long> gearIds = applicationInfo.getGearIds();
+    List<Gear> gears = (gearIds == null)? Collections.emptyList() : gearRepository.findAllById(gearIds);
+
+    return PracticeApplicationResponse.from(applicationInfo, gears);
+  }
+
+  @Transactional
+  public void cancelApplication(Long applicationId, String userId) {
+//    int affected = practiceApplicationRepository.deleteByIdAndUserId(practiceApplicationId, userId);
+//    if(affected == 0){
+//      throw new EntityNotFoundException("신청 내역이 없거나 이미 삭제됨");
+//    }//DELETE 로직
+
+    PracticeApplication app = practiceApplicationRepository.findById(applicationId)
+        .orElseThrow(() -> new EntityNotFoundException("신청 내역 없음"));
+    if (!userId.equals(app.getUserId())) {
+      throw new IllegalArgumentException("본인 신청만 취소할 수 있습니다.");
+    }
+    if(app.getStatus() == Status.PAYMENT_CONFIRMED) throw new RuntimeException("확정 이후 취소 불가");
+    app.setStatus(Status.CANCELLED);
+  }
+
+  @Transactional
+  public void changeStatus(Long applicationId, int statusCode) {
+    PracticeApplication application = practiceApplicationRepository.findById(applicationId)
+        .orElseThrow(()-> new EntityNotFoundException("유효하지 않는 신청"));
+    if(application.getStatus() == Status.PAYMENT_CONFIRMED){
+      throw new IllegalArgumentException("확정 후 변경 불가");
+    }
+    PracticeApplication.Status status = PracticeApplication.Status.fromCode(statusCode);
+    application.setStatus(status);
```

Comment:
```suggestion
    application.setStatus(PracticeApplication.Status.fromCode(statusCode));
```
불필요한 변수 선언은 없어도 될것 같습니다